odoo/addons/base/models/ir_binary.py

```diff
@@ -45,14 +45,9 @@ def _find_record(
         if not record:
             raise MissingError(f"No record found for xmlid={xmlid}, res_model={res_model}, id={res_id}")
 
-        record = self._find_record_check_access(record, access_token, field)
-        return record
-
-    def _find_record_check_access(self, record, access_token, field):
-        if record._name == 'ir.attachment':
-            return record.validate_access(access_token)
-
-        record.check_access('read')
+        if record._can_elevate_access(access_token, field):
+            record = record.sudo()
+        record.check_access("read")
```

Comment:
Don't we want to call `_can_elevate_access` only if standard read check fails instead?
And then when using sudo there's no point to check read again.